compiler/AST/AggregateType.cpp

```diff
@@ -2224,9 +2226,124 @@ void AggregateType::buildDefaultInitializer() {
   }
 }
 
+static bool hasFullyGenericField(AggregateType* at) {
+  for_fields(field, at) {
+    DefExpr* defExpr = field->defPoint;
+    if (!field->hasFlag(FLAG_TYPE_VARIABLE) && !field->hasFlag(FLAG_PARAM) &&
+        !field->hasFlag(FLAG_SUPER_CLASS)) {
```

Comment:
This code doesn't handle a field like `var x: integral;`. Should it?

Reply:
I'm working on a commit that will handle this case, and adding a couple of tests.